### DISCUSSION TITLE
Set watch callback event type when calling setValue

### DISCUSSION THREAD

**Discussion by BrendanC23** [react-hook-form/react-hook-form#12197] [Ideas]
The `form.watch(callback)` function has an event type parameter, which is usually set to `change` for user-triggered inputs. When the form is updated using `setValue`, the type is `undefined`. I would like to be able to pass a custom change type to `setValue` and `reset`.

There is a similar issue from 2022: https://github.com/react-hook-form/react-hook-form/issues/9363 that I would like to bring attention to.

`setValue('fieldName', newValue, { changeType: 'newChangeTypeOverride' });`

**Comment by regileeso** [comment]
I'd give this one +1

My situation is the following. I'm using Redux to keep track of the changes inside react-hook-form. Also, when initializing the form, the values come from Redux selectors. _watch()_ is used to dispatch any changes to Redux. This all works fine.

Now, the values inside Redux may be changed from another component by dispatching an action. I then update the form values using _useSelector()_ and _setValue()_ and I need to avoid _watch()_ dispatching the data again to Redux by checking _type_. 

But yes, _type_ is undefined when I use _setValue()_.

Is there any way to achieve that? Currently it just results in an endless hook loop...

**Reply by cike8899** [reply]
I also encountered the same confusion.
